dask_geomodeling/geometry/text.py

```diff
@@ -125,6 +116,9 @@ def parser(description):
 
         # Assign the extra columns one by one. This preserves categoricals.
         for name in extra_columns_aligned.columns:
-            f[name] = extra_columns_aligned[name]
+            if extra_columns_aligned[name].isnull().all():
```

Comment:
The comment before the loop about preserving categoricals seems to be misplaced now.